packages/compass-context-menu/src/use-context-menu.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useContext, useMemo, useRef } from 'react';
+import { Context } from './context-menu-provider';
+import { appendContextMenuContent } from './context-menu-content';
+import type { MenuItem } from './types';
+
+/**
+ * @returns an object with methods to {@link register} content for the menu and {@link close} the menu
+ */
+export function useContextMenu({
+  Menu,
+}: {
+  Menu: React.ComponentType<{
```

Comment:
I guess we could keep it headless by letting it accept a component?

Open to a cleaner solution